internal/proxy/proxy_config.go

```diff
@@ -81,6 +82,7 @@ type OptionsConfig struct {
 	SkipAuthRegex   []string          `yaml:"skip_auth_regex"`
 	AllowedGroups   []string          `yaml:"allowed_groups"`
 	TLSSkipVerify   bool              `yaml:"tls_skip_verify"`
+	PreserveHost    bool              `yaml:"preserve_host"`
```

Comment:
might be good to add a config unit test as well

Reply:
Done.